source/vibe/core/net.d

```diff
@@ -193,11 +195,15 @@ TCPConnection connectTCP(NetworkAddress addr, NetworkAddress bind_address = anyA
 		scope uaddr = new RefAddress(addr.sockAddr, addr.sockAddrLen);
 		scope baddr = new RefAddress(bind_address.sockAddr, bind_address.sockAddrLen);
 
-		// FIXME: make this interruptible
 		auto result = asyncAwaitUninterruptible!(ConnectCallback,
-			cb => eventDriver.sockets.connectStream(uaddr, baddr, cb)
-			//cb => eventDriver.sockets.cancelConnect(cb)
-		);
+			cb => eventDriver.sockets.connectStream(uaddr, baddr, cb),
+			(ConnectCallback cb, StreamSocketFD sock_fd) {
+				eventDriver.sockets.cancelConnectStream(sock_fd);
+			}
+		)(timeout);
+
+		if (result[1] == ConnectStatus.cancelled)
+			result[1] = ConnectStatus.timeout;
```

Comment:
`asyncAwaitUninterruptible` could be changed to `asyncAwait`, now that cancellation is possible. The `if` condition above would then have to be `(!result.completed)`, wich will also work with the updated callback semantics for the eventcore PR.